### PR TITLE
Search: Add animation to modal summoning

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:
* Slides in the search modal from bottom of the screen upon summoning.

Large GIF (1788x808)

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with a Jetpack Search subscription.
* Navigate to `/` and enter a query into a search input. Verify that the search modal appears with an entry animation.
* Navigate to `/?s=some-search-value`. Verify that the search modal appears with an entry animation.

Note that we currently have special handling when dismissing the search modal if the initial href includes `/?s=`. Dismissing the modal in this instance will strip all search-related query string values and navigate the window to the resulting href (e.g. `/`).

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/instant-search/lib/styles/_animations.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		transform: translate3d( 0, 0, 0 );
+	}
+
+	to {
+		transform: translate3d( 0, 100vh, 0 );
+		visibility: hidden;
+	}
+}
+
+@keyframes fadeIn {
```
Comment (simison) [inline projects/plugins/jetpack/modules/search/instant-search/lib/styles/_animations.scss:23]: Hi! These end up in global scope in CSS, so anyone else (like Gutenberg or Jetpack itself) defining the same keyframe ends up overwriting it. You'll potentially get subtle bugs depending on CSS loading order.

https://github.com/Automattic/jetpack/search?l=SCSS&q=fadeIn

Recommend to prefix with something like `jetpack_search_`

Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/instant-search/lib/styles/_animations.scss]: Hi @simison, thanks for the heads up! This has now been fixed.

**Review by bluefuton — COMMENTED** [review]
Nice - I like the gentle introduction of the modal. The opening is nice and smooth.

When I close the modal it's a bit flickery (both in Firefox and Chrome):

**Comment by jsnmoon** [comment]
> When I close the modal it's a bit flickery (both in Firefox and Chrome) (@bluefuton)

Fixed! This is due to the fact that we're trying to apply an opacity transition on the overlay, which  includes the search results as its child. Solved via `transition-delay`.

**Review by bluefuton — COMMENTED** [review]
Flickering is sorted! Thanks :)

I'm seeing some opacity weirdness on the modal close. As shown in the GIF, only part of the underlying page seems to do the fade (with the header and footer immediately returning to their original colour):



(This is in FF Mac.)

**Comment by jsnmoon** [comment]
> I'm seeing some opacity weirdness on the modal close. As shown in the GIF, only part of the underlying page seems to do the fade (with the header and footer immediately returning to their original colour)

Fixed; this was happening due to specifying a low z-index for the hidden overlay.

**Comment by bluefuton** [comment]
> Fixed; this was happening due to specifying a low z-index for the hidden overlay.

@jsnmoon did you push the commit for this fix? I'm still seeing the same behaviour 🤔

**Comment by jsnmoon** [comment]
> @jsnmoon did you push the commit for this fix? I'm still seeing the same behaviour 🤔

😆 Just pushed!

**Review by bluefuton — COMMENTED** [review]
Works really well on Mac Firefox + Safari now 👏 The opacity issue is fixed.

I found some odd scroll behaviour on our old friends IE11 and Edge. When the modal opens with the animation, the modal is scrolled down slightly meaning that the search input and form controls are hidden. I checked against the main branch and it seems to work as expected there.

Edge:
Screen Shot 2021-03-29 at 14 43 51

IE11:
Screen Shot 2021-03-29 at 14 44 43

I tested in local IE VMs (hard to see the animation properly on Browserstack).

**Comment by jsnmoon** [comment]
> I found some odd scroll behaviour on our old friends IE11 and Edge. When the modal opens with the animation, the modal is scrolled down slightly meaning that the search input and form controls are hidden. I checked against the main branch and it seems to work as expected there.

This was due to IE11 not supporting `element.scrollTo`; I went ahead and introduced a fallback using `element.scrollTop = 0`.

I'm not sure why Edge would be failing, given that it's just a Chrome wrapper now. Which version of Edge are you testing against?

**Review by bluefuton — APPROVED** [review]
Looks good! Tested in Firefox, Safari, IE11 and Edge.

> I'm not sure why Edge would be failing, given that it's just a Chrome wrapper now. Which version of Edge are you testing against?

It was a fairly vintage version in my VM (v44). Just against that and also v89 on Browserstack and both work well. ✅

**Comment by jsnmoon** [comment]
Shipped in r223517-wpcom.

**Review by jeherve — APPROVED** [review]
Fancy! It works well for me, merging.

**Pull request merged**